src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,83 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+    private static final String COMMA = ",";
+    private static final String NEW_LINE = "\n";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] fileData = readFile(fromFileName).split(NEW_LINE);
+        String[] resultData = prepareStingToWrite(fileData);
+        clearFile(toFileName);
+        writeData(resultData, toFileName);
+    }
+
+    private String readFile(String fromFileName) {
+        String content = "";
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                content += line + NEW_LINE;
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file: " + fromFileName, e);
+        }
+        return content;
+    }
+
+    private String[] prepareStingToWrite(String[] notGroupedData) {
+        int supplySum = 0;
+        int buySum = 0;
+        int addition = 0;
+        for (int i = 0; i < notGroupedData.length; i++) {
+            String[] splitedArray = notGroupedData[i].split(COMMA);
+            addition = Integer.parseInt(splitedArray[1]);
+            if (splitedArray[0].equals(SUPPLY)) {
+                supplySum += addition;
+            }
+            if (splitedArray[0].equals(BUY)) {
+                buySum += addition;
+            }
+        }
+        return new String[]{SUPPLY + COMMA + supplySum,
+                BUY + COMMA + buySum,
+                RESULT + COMMA + (supplySum - buySum)};
+    }
+
+    private void clearFile(String fileName) {
```

Comment:
The `clearFile` method deletes and recreates the file. While this works, it might not be the most efficient way to clear a file. Instead, you could open the file with FileWriter and set its content to an empty string.